datasets/tf_datasets.py

```diff
@@ -0,0 +1,117 @@
+"""
+https://www.tensorflow.org/datasets
+"""
+
+
+from typing import Optional
+from sisyphus import *
+
+
+class DownloadAndPrepareTfDatasetJob(Job):
+    """
+    https://www.tensorflow.org/datasets
+
+        pip install tensorflow-datasets
+
+    It further needs some extra dependencies, for example for 'librispeech':
+
+        pip install apache_beam
+        pip install pydub
+        # ffmpeg installed
+
+    See here for some more:
+    https://github.com/tensorflow/datasets/blob/master/setup.py
+
+    Also maybe::
+
+        pip install datasets  # for Huggingface community datasets
+    """
+
+    def __init__(
+        self,
+        dataset_name: str,
+        *,
+        max_simultaneous_downloads: Optional[int] = None,
```

Comment:
I am not sure if it is a good idea to start mixing typing styles (python types vs docstring types). @curufinwe opinions on this?

Reply:
We used docstring types in RETURNN only because we wanted to have it Python 2 compatible. Also, I think PyCharm is the only tool which really makes use of them. Otherwise it's a bad idea to use docstring types and you always should use the standard type hints, where there are also other tools who can make use of them.

Reply:
It is clear that using standard types is better, but this is about if we want it consistent or not.

Reply:
I would say that we also should change it everywhere else to standard types (but this can be done gradually), and even disallow any new code with docstring types.